internal/services/amlfilesystem/aml_file_system_resource.go

```diff
@@ -0,0 +1,518 @@
+	identityValue, err := identity.ExpandUserAssignedMapFromModel(input)
+	if err != nil {
+		return nil, fmt.Errorf("expanding `identity`: %+v", err)
+	}
+
+	output := amlfilesystems.AmlFilesystemIdentity{
+		Type: pointer.To(amlfilesystems.AmlFilesystemIdentityType(string(identityValue.Type))),
+	}
+
+	if identityValue.Type == identity.TypeUserAssigned {
+		output.UserAssignedIdentities = pointer.To(make(map[string]amlfilesystems.UserAssignedIdentitiesProperties))
+		for k := range identityValue.IdentityIds {
+			(*output.UserAssignedIdentities)[k] = amlfilesystems.UserAssignedIdentitiesProperties{}
+		}
+	}
+
+	return &output, nil
+}
+
+func flattenAMLFileSystemIdentity(input *amlfilesystems.AmlFilesystemIdentity) ([]identity.ModelUserAssigned, error) {
+	if input == nil {
+		return nil, nil
+	}
+
+	identityIds := make(map[string]identity.UserAssignedIdentityDetails, 0)
+	for k, v := range *input.UserAssignedIdentities {
+		identityIds[k] = identity.UserAssignedIdentityDetails{
+			ClientId:    v.ClientId,
+			PrincipalId: v.PrincipalId,
+		}
+	}
+
+	identityValue := identity.UserAssignedMap{
+		Type:        identity.Type(string(pointer.From(input.Type))),
+		IdentityIds: identityIds,
+	}
+
+	output, err := identity.FlattenUserAssignedMapToModel(&identityValue)
+	if err != nil {
+		return nil, fmt.Errorf("expanding `identity`: %+v", err)
+	}
+
+	return *output, nil
+}
+
+func expandAMLFileSystemMaintenanceWindowForCreate(input []MaintenanceWindow) amlfilesystems.AmlFilesystemPropertiesMaintenanceWindow {
+	maintenanceWindow := &input[0]
+
+	return amlfilesystems.AmlFilesystemPropertiesMaintenanceWindow{
+		DayOfWeek:    pointer.To(maintenanceWindow.DayOfWeek),
+		TimeOfDayUTC: pointer.To(maintenanceWindow.TimeOfDayInUTC),
+	}
+}
+
+func expandAMLFileSystemMaintenanceWindowForUpdate(input []MaintenanceWindow) *amlfilesystems.AmlFilesystemUpdatePropertiesMaintenanceWindow {
+	if len(input) == 0 {
+		return nil
+	}
+
+	maintenanceWindow := &input[0]
+	result := amlfilesystems.AmlFilesystemUpdatePropertiesMaintenanceWindow{}
+
+	if v := maintenanceWindow.DayOfWeek; v != "" {
+		result.DayOfWeek = pointer.To(v)
+	}
+
+	if v := maintenanceWindow.TimeOfDayInUTC; v != "" {
+		result.TimeOfDayUTC = pointer.To(v)
+	}
+
+	return &result
+}
+
+func flattenAMLFileSystemMaintenanceWindow(input amlfilesystems.AmlFilesystemPropertiesMaintenanceWindow) []MaintenanceWindow {
+	var result []MaintenanceWindow
+
+	maintenanceWindow := MaintenanceWindow{
+		DayOfWeek:      pointer.From(input.DayOfWeek),
+		TimeOfDayInUTC: pointer.From(input.TimeOfDayUTC),
+	}
+
+	return append(result, maintenanceWindow)
+}
+
+func expandAMLFileSystemKeyEncryptionKey(input []KeyEncryptionKey) *amlfilesystems.AmlFilesystemEncryptionSettings {
+	if len(input) == 0 {
+		return nil
+	}
+
+	keyEncryptionKey := &input[0]
+
+	result := &amlfilesystems.KeyVaultKeyReference{
+		KeyUrl: keyEncryptionKey.KeyUrl,
+		SourceVault: amlfilesystems.KeyVaultKeyReferenceSourceVault{
```

Comment:
Persumably, we only need to expose the key url here as the sourceVault can be deduced from the key URL?

Reply:
I align with [the existing pattern](https://github.com/hashicorp/terraform-provider-azurerm/blob/864e6b46b8feb146736b036ea828a2e2ae60f798/internal/services/compute/encryption_settings.go#L97).